ext/random/randomizer.c

```diff
@@ -413,26 +414,13 @@ PHP_METHOD(Random_Randomizer, getBytesFromString)
 			ZSTR_VAL(retval)[total_size++] = ZSTR_VAL(source)[offset];
 		}
 	} else {
-		uint64_t mask;
-		if (source_length <= 0x1) {
-			mask = 0x0;
-		} else if (source_length <= 0x2) {
-			mask = 0x1;
-		} else if (source_length <= 0x4) {
-			mask = 0x3;
-		} else if (source_length <= 0x8) {
-			mask = 0x7;
-		} else if (source_length <= 0x10) {
-			mask = 0xF;
-		} else if (source_length <= 0x20) {
-			mask = 0x1F;
-		} else if (source_length <= 0x40) {
-			mask = 0x3F;
-		} else if (source_length <= 0x80) {
-			mask = 0x7F;
-		} else {
-			mask = 0xFF;
-		}
+		uint64_t mask = max_offset;
+		// Copy the top-most bit into all lower bits.
+		// Shifting by 4 is sufficient, because max_offset
+		// is guaranteed to be an 8-bit integer.
```

Comment:
I'm probably just being dumb, but why is ``max_offset`` and 8-bit integer? AFAIK ``size_t`` has no such requirement

Reply:
The else-path is only taken for values less than or equal to 255.

Reply:
I've rephrased the comment a little to make this clearer.

Reply:
*duh* yeah okay I was just being dumb. Thank for clarifying!